File: Front End Angular/src/app/services/film.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {InsertFilmDTO} from '../dto/film/request/insert-film-dto';
import {finalize, Observable} from 'rxjs';
import {ResponseFilmDTO} from '../dto/film/response/response-film-dto';
import {EditFilmDTO} from '../dto/film/request/edit-film-dto';
import {LongOmdbResponseApiDto} from '../dto/omdbapi/response/long-omdb-response-api-dto';
import {SharedService} from "./shared.service";

@Injectable({
  providedIn: 'root'
})
export class FilmService {
  private BASE_URL = 'http://localhost:8080';

  constructor(
      private http: HttpClient,
      private sharedService: SharedService
  ) {}


  insert(dto: InsertFilmDTO): Observable<ResponseFilmDTO> {
    return this.http.post<ResponseFilmDTO>(`${this.BASE_URL}/staff/film`, dto);
  }


  findAll(): Observable<ResponseFilmDTO[]> {
    return this.http.get<ResponseFilmDTO[]>(`${this.BASE_URL}/film`);
  }


  findById(id: number): Observable<ResponseFilmDTO> {
    return this.http.get<ResponseFilmDTO>(`${this.BASE_URL}/film/${id}`);
  }


  findByIdGenere(idGenere: number): Observable<ResponseFilmDTO[]> {
    return this.http.get<ResponseFilmDTO[]>(
      `${this.BASE_URL}/film/genere/${idGenere}`
    );
  }


  findByTitolo(titolo: string): Observable<LongOmdbResponseApiDto[]> {
    this.sharedService.loading = true;
    return this.http.get<LongOmdbResponseApiDto[]>(
      `${this.BASE_URL}/film/titolo`,
      { params: { titolo } }
    ).pipe(
        finalize(() => this.sharedService.loading = false)
    );
  }


  editById(id: number, dto: EditFilmDTO): Observable<ResponseFilmDTO> {
    return this.http.patch<ResponseFilmDTO>(
      `${this.BASE_URL}/staff/film/${id}`,
      dto
    );
  }


  removeById(id: number): Observable<void> {
    return this.http.delete<void>(`${this.BASE_URL}/staff/film/${id}`);
  }
  existsByImdbId(id: string): Observable<boolean> {
    return this.http.get<boolean>(`${this.BASE_URL}/staff/film/check-exists`, {
      params: { id: id }
    });
  }

}

File: Back End Spring/src/main/java/org/elis/movieexplorer/service/definition/FilmService.java
package org.elis.movieexplorer.service.definition;

import java.util.List;

import org.elis.movieexplorer.dto.film.request.EditFilmDTO;
import org.elis.movieexplorer.dto.film.request.InsertFilmDTO;
import org.elis.movieexplorer.dto.film.response.ResponseFilmDTO;
import org.elis.movieexplorer.dto.omdbapi.response.LongOmdbResponseApiDTO;

public interface FilmService {
	// CREATE
	ResponseFilmDTO insert(InsertFilmDTO dto);
	
	// READ
	List<ResponseFilmDTO> findAll();
	
	ResponseFilmDTO findById(Long id);
	
	List<ResponseFilmDTO> findByIdGenere(Long id);
	
	// UPDATE
	ResponseFilmDTO editById(Long id, EditFilmDTO fMod);
	
	// DELETE
	void removeById(Long id);

    List<LongOmdbResponseApiDTO> findByTitolo(String titolo);

	Boolean findByImdbID(String id);
}
